### PR TITLE
 tests(crypto): Increase coverage and mark private methods

### PR DESCRIPTION
**Pull request by JeremiGendron**
## Proposed changes
 - document validate rules param correct types (Object|Function|String)
 - __validateWith<> methods are now private
 - tests for __validateWith<> changed to validate("...", <>)
 - increase coverage for errors() function
 - modify validate to check for "typeof rules === 'string'" (more below)

Refs: #1782 #1775

The validate function was checking for `rules instanceof String`, which never evaluates to true for a string literal. Changed to `typeof rules === 'string'`

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [x] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/crypto/src/validation/validator.ts [hunk 1]
```diff
@@ -108,7 +115,7 @@ export class Validator {
      * @param  {String} rule
      * @return {void}
      */
-    public __validateWithJoi(attributes, rules) {
+    private __validateWithJoi(attributes, rules) {
```
Comment (faustbrian) [inline packages/crypto/src/validation/validator.ts]: The `__` prefix has to be removed, if you read the issue.

File: packages/crypto/src/validation/validator.ts [hunk 2]
```diff
@@ -76,13 +76,20 @@ export class Validator {
...
+    }
+
     /**
      * Run the validator's rules against its data using a rule.
      * @param  {*} attributes
      * @param  {String} rule
      * @return {void}
      */
-    public __validateWithRule(attributes, rules) {
+    private __validateWithRule(attributes, rules) {
```
Comment (faustbrian) [inline packages/crypto/src/validation/validator.ts]: The `__` prefix has to be removed, if you read the issue.

File: packages/crypto/src/validation/validator.ts [hunk 3]
```diff
@@ -98,7 +105,7 @@ export class Validator {
      * @param  {String} rule
      * @return {void}
      */
-    public __validateWithFunction(attributes, validate) {
+    private __validateWithFunction(attributes, validate) {
```
Comment (faustbrian) [inline packages/crypto/src/validation/validator.ts]: The `__` prefix has to be removed, if you read the issue.

File: packages/crypto/src/validation/validator.ts [hunk 4]
```diff
@@ -76,13 +76,20 @@ export class Validator {
         this.rules[name] = implementation;
     }
 
+    /**
+     * Reset any previous results.
+     */
+    public __reset() {
```
Comment (faustbrian) [inline packages/crypto/src/validation/validator.ts]: Public methods have no prefixes, if you read the issue.

**Comment by JeremiGendron** [comment]
@faustbrian thank you

File: packages/crypto/src/validation/validator.ts (hunk 4 above)
Comment (JeremiGendron) [reply packages/crypto/src/validation/validator.ts]: branch updated

**Comment by faustbrian** [comment]
The core-api changes should not be included in this PR.

**Comment by JeremiGendron** [comment]
@faustbrian not sure if `git revert 0abbee01` was the right move

**Pull request merged**